terraform/batch-processing/s3.tf

```diff
@@ -0,0 +1,23 @@
+locals {
+    // Flag so we can force delete s3 buckets with items in for pr and shortcode environments only.
+    is_temp = length(regexall("[a-z]{2,4}-?[0-9]{2}", var.environment)) > 0
```

Comment:
Nice one. It matches both pr and shortcode 🚀 . The only thing is that last {2}. It won't match multiple digits in a pr like pr-123. `[a-z]{2,4}-?[0-9]+` I think will work. I didn't test it though.